compiler/src/dmd/e2ir.d

```diff
@@ -3127,6 +3076,27 @@ elem* toElem(Expression e, IRState *irs)
     elem* visitCall(CallExp ce)
     {
         //printf("[%s] CallExp.toElem('%s') %p, %s\n", ce.loc.toChars(), ce.toChars(), ce, ce.type.toChars());
+        /* Do this check during code gen rather than semantic() because concatenation is
+         * allowed in CTFE, and cannot distinguish that in semantic().
```

Comment:
Two major issues with this:
- GDC and LDC have to implement this error check themselves (is there a testcase for it?)
- `-o-` will not trigger this error

Reply:
> GDC and LDC have to implement this error check themselves (is there a testcase for it?)

This check was previously in `visitCat()`, so this PR isn't making things worse. And the reason it's here is explained in the comment. Do you know of a better place where to verify this?

> `-o-` will not trigger this error
Why would it? AFICT `-o-` simply skips the generation of the `.o` file. How is this connected to string concatenation with `-betterC`?

Reply:
`-o-` skips codegen, thus skips `e2ir` ?

Reply:
looks like you are right that this PR does not make the situation worse.

Reply:
> `-o-` skips codegen, thus skips `e2ir`

And doing so also skips the rest of the backend. Therefore, the `CatExp` that would otherwise be leaked to the glue layer is ignored. Because code generation is omitted, the leak is silent as it doesn't cause any trouble.